metadata_please/sdist.py

```diff
@@ -30,12 +30,17 @@ def from_zip_sdist(zf: ZipFile) -> bytes:
 def basic_metadata_from_zip_sdist(zf: ZipFile) -> BasicMetadata:
     requires = [f for f in zf.namelist() if f.endswith("/requires.txt")]
     requires.sort(key=len)
-    if not requires:
-        return BasicMetadata((), frozenset(), "-")
+    if requires:
+        requires_data = zf.read(requires[0])
+        assert requires_data is not None
+    else:
+        requires_data = b""
 
-    data = zf.read(requires[0])
-    assert data is not None
-    return BasicMetadata.from_sdist_pkg_info_and_requires(b"", data)
+    pkg_info = next(f for f in zf.namelist() if f.endswith("/PKG-INFO"))
```

Comment:
Consider sorting this by length and taking the shortest; these might exist in test data and these archives are generally alphabetic order.  I'll try to find you a real-world artifact that needs this.

Reply:
nose 1.3.1 (test data)
ucxdsa 2024.4.23 (project rename, didn't clean)
relaxed-poetry 0.2.2 (test data)
wish 1.0.1 (`.lib` dir)

additionally

rfernet 0.1.3 (no top-level dir)